### DISCUSSION TITLE
Add "Vomiting" as a guest status

### DISCUSSION THREAD

**Discussion by tellovishous** [OpenRCT2/OpenRCT2#15737] [Feature requests]
Recently thought of a small feature request:
Could we add the status "Vomiting" to show up in the guest window when a guest vomits? The window will describe nearly all the other actions for guests, yet when sick guests stop to barf, it'll still say "Walking" even though they aren't. I know it's pretty minute, but I think maybe we could add the status for the sake of descriptive accuracy.

**Comment by AaronVanGeffen** [comment]
You could change `Peep::FormatActionTo` to return a different string when a guest is vomiting. Trouble is, right now vomiting is not a _state_ (which is what that function is checking for), but an action. You could hack in a check like `Action == PEEP_ACTION_THROW_UP`, though. A similar check appears to be present to check whether a guest is drowning.

**Comment by ocalhoun6** [comment]
This _could_ actually have some small practical use, not just an aesthetic thing: If you had the peeps list open and set to show only guests who were vomiting, _and_ you had the park map open, it would show you in real time where vomit is happening, so you'd know where needs to be cleaned.

This would let you be a bit more proactive about it, rather than waiting for peeps to complain that the path is disgusting, which is what you'd have to do now to get a similar effect. (Looking for guests with that thought also is less precise, because guests wander around while thinking that thought and can get a significant distance from where the vomit actually is before the thought goes away.)

**Comment by tellovishous** [comment]
ocalhoun6 makes a good point actually, perhaps it could still have some use outside of aesthetics. And it sounds like it could be possible to add.

**Comment by BryanDiLaura** [comment]
Sounds interesting and could be useful; I'd be willing to take a stab at it if there's interest!

**Comment by tupaschoal** [comment]
Well, you can give it a try and we can discuss over at the draft pr or on discord
